drivers/soundwire/intel.c

```diff
@@ -1044,6 +1044,7 @@ intel_hw_free(struct snd_pcm_substream *substream, struct snd_soc_dai *dai)
 
 	dma->hw_params = NULL;
 	dma->pdi = NULL;
+	kfree(dma->stream->name);
```

Comment:
unrelated to this patch: `sdw_release_stream()` below just does a `kfree(stream)` so after it dma->stream contains an invalid pointer. Does the `dma` object get freed after that too? If not, maybe it would be good to set `dma->stream = NULL;` after calling `sdw_release_stream()`

Reply:
Hmm, @lyakh you raise a good point. startup() should be a pair with shutdown() and prepare() with hw_free(). So if you follow that, the stream (and the name) should not be free here, but rather in shutdown.

Reply:
the 'name' is not allocated as part of the stream allocation